src/datasources/db/v2/database-migrator.service.spec.ts

```diff
@@ -0,0 +1,140 @@
+import { Test, type TestingModule } from '@nestjs/testing';
+import { ConfigModule } from '@nestjs/config';
+import { ConfigurationModule } from '@/config/configuration.module';
+import configuration from '@/config/entities/__tests__/configuration';
+import { TestLoggingModule } from '@/logging/__tests__/test.logging.module';
+import { PostgresDatabaseService } from '@/datasources/db/v2/postgres-database.service';
+import { DatabaseMigrator } from '@/datasources/db/v2/database-migrator.service';
+import {
+  LoggingService,
+  type ILoggingService,
+} from '@/logging/logging.interface';
+import type { DataSource } from 'typeorm';
+import { TestPostgresDatabaseModuleV2 } from '@/datasources/db/v2/test.postgres-database.module';
+
+describe('PostgresDatabaseService', () => {
+  let moduleRef: TestingModule;
+  let postgresDatabaseService: PostgresDatabaseService;
+  let databaseMigratorService: DatabaseMigrator;
+  let loggingService: ILoggingService;
+  let connection: DataSource;
+  const NUMBER_OF_RETRIES = 2;
+  const truncateLockQuery = 'TRUNCATE TABLE "_lock";';
+  const insertLockQuery = 'INSERT INTO "_lock" (status) VALUES ($1);';
+
+  beforeAll(async () => {
+    // We should not require an SSL connection if using the database provided
+    // by GitHub actions
+    const isCIContext = process.env.CI?.toLowerCase() === 'true';
+    const baseConfiguration = configuration();
+    const testConfiguration: typeof configuration = () => ({
+      ...baseConfiguration,
+      db: {
+        ...baseConfiguration.db,
+        migrator: {
+          numberOfRetries: NUMBER_OF_RETRIES,
+          migrationsExecute: false,
+          retryAfter: 10,
+        },
+        connection: {
+          postgres: {
+            ...baseConfiguration.db.connection.postgres,
+            ssl: {
+              ...baseConfiguration.db.connection.postgres.ssl,
+              enabled: !isCIContext,
+              caPath: join(__dirname, '../../../../db_config/test/server.crt'),
+            },
+          },
+        },
+      },
+    });
+
+    moduleRef = await Test.createTestingModule({
+      imports: [
+        TestLoggingModule,
+        ConfigurationModule.register(testConfiguration),
+        ConfigModule,
+        TestPostgresDatabaseModuleV2,
+      ],
+      providers: [DatabaseMigrator],
+    }).compile();
+
+    databaseMigratorService = moduleRef.get<DatabaseMigrator>(DatabaseMigrator);
+    postgresDatabaseService = moduleRef.get<PostgresDatabaseService>(
+      PostgresDatabaseService,
+    );
+    loggingService = moduleRef.get<ILoggingService>(LoggingService);
+
+    connection = postgresDatabaseService.getDataSource();
+  });
+
+  afterAll(async () => {
+    await moduleRef.close();
+  });
+
+  beforeEach(() => {});
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('migrate()', () => {
+    it('Should log the start and end of the migration process', async () => {
+      jest.spyOn(loggingService, 'info');
```

Comment:
I am impartial regarding this testing method, but we'd generally use a mock logging service consisting of jest mocks. That way we don't need to `spyOn` the service.

Reply:
Changed for the sake of consistency
https://github.com/safe-global/safe-client-gateway/commit/2ea0e5d8a752fd31cff745ff3490d8fe8134ddc7